### PR TITLE
Add single line description of ExecutionPlan (#2216)

### LINKED ISSUE #2216: Render Single Line Description of ExecutionPlan
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

In IOx we have a [`one_line`](https://github.com/influxdata/influxdb_iox/blob/main/query/src/exec/query_tracing.rs#L258) function that can be used to render a single line description of an `ExecutionPlan`. This can be very useful for debugging where the full recursive plan printout would information overload

**Describe the solution you'd like**

Upstream this code to DataFusion

**Describe alternatives you've considered**

We could not

**Additional context**

I used this within the POC of #2199

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Closes #2216

 # Rationale for this change

See ticket

# What changes are included in this PR?

Adds the ability to get a single-line summary of the root of an ExecutionPlan

# Are there any user-facing changes?

No

### CODE REVIEW

File: datafusion/core/src/physical_plan/display.rs [hunk 1]
```diff
@@ -101,6 +101,32 @@ impl<'a> DisplayableExecutionPlan<'a> {
             show_metrics: self.show_metrics,
         }
     }
+
+    /// Return a single-line summary of the root of the plan
+    pub fn one_line(&self) -> impl fmt::Display + 'a {
```
Comment (alamb) [inline datafusion/core/src/physical_plan/display.rs:106]:
```suggestion
    // Example: `ProjectionExec: expr=[a@0 as a]`.
    pub fn one_line(&self) -> impl fmt::Display + 'a {
```

File: datafusion/core/src/physical_plan/display.rs [hunk 2]
```diff
@@ -101,6 +101,33 @@ impl<'a> DisplayableExecutionPlan<'a> {
             show_metrics: self.show_metrics,
         }
     }
+
+    /// Return a single-line summary of the root of the plan
+    // Example: `ProjectionExec: expr=[a@0 as a]`.
```
Comment (tustvold) [inline datafusion/core/src/physical_plan/display.rs:106]:
```suggestion
    /// Example: `ProjectionExec: expr=[a@0 as a]`.
```

**Comment by yjshen** [comment]
Another physical plan description optimization would be to limit each physical plan's max line width. This would be especially useful for the ParquetExec node:

```
ProjectionExec: expr=[ss_sold_date_sk@0 as ss_sold_date_sk, ss_customer_sk@1 as ss_customer_sk]
      FilterExec: ss_sold_date_sk@0 IS NOT NULL
        ParquetExec: limit=None, partitions=[/Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00000-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00023-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00012-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00001-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00024-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00030-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00008-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00056-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00010-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00078-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00028-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00016-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00017-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00002-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00021-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00027-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00032-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00031-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00034-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00014-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00005-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00026-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00019-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00011-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00013-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00089-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00020-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00029-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00004-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00015-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00007-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00018-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00067-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00022-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00009-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00006-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00045-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00033-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00003-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00025-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00035-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00096-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00062-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00068-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00066-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00072-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00077-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00086-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00061-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00063-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00052-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00082-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00075-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00087-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00059-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00070-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00053-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00050-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00084-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00085-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00049-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00080-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00038-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00046-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00074-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00043-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00076-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00093-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00058-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00039-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00090-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00036-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00083-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00051-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00079-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00092-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00055-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00081-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00088-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00071-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00044-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00042-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00069-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00048-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00054-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00095-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00065-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00047-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00094-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00037-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00073-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00064-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00057-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00041-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00060-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00091-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00040-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet, /Users/shenyijie/oss/spark-blaze-extension/dev/tpcds_1g/store_sales/part-00097-8ecba12f-a127-4bca-8d50-f9516808f050-c000.snappy.parquet], predicate=true, projection=[ss_sold_date_sk, ss_customer_sk]
```

It would be wonderful if I could reduce this to:
```
ParquetExec: limit=None, predicate=true, projection=[ss_sold_date_sk, ss_customer_sk], partitions=[100 items ... ]
```

**Comment by tustvold** [comment]
Perhaps #2079 might help with that :sweat_smile:

**Pull request merged**